modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/configuration/permission/propagation/DLFolderPortletConfigurationPermissionPropagation.java

```diff
@@ -56,67 +44,28 @@ public String getMessage() {
 	}
 
 	@Override
-	public boolean getPermissionPropagation(PortletRequest portletRequest) {
-		ThemeDisplay themeDisplay = (ThemeDisplay)portletRequest.getAttribute(
-			WebKeys.THEME_DISPLAY);
-
-		long groupId = getGroupId(portletRequest);
-
-		long classPK = getClassPK(portletRequest);
-
-		if (classPK == groupId) {
-			classPK = DLFolderConstants.DEFAULT_PARENT_FOLDER_ID;
-		}
-
-		PermissionPropagationEntry permissionPropagationEntry =
-			_permissionPropagationEntryLocalService.
-				fetchPermissionPropagationEntry(
-					themeDisplay.getCompanyId(), groupId,
-					DLFolder.class.getName(), classPK);
-
-		if (permissionPropagationEntry != null) {
-			return permissionPropagationEntry.isPropagation();
-		}
-
-		if (classPK == DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
-			return true;
-		}
-
-		return false;
+	protected String getClassName() {
+		return DLFolder.class.getName();
 	}
 
 	@Override
-	public boolean isShow(PortletRequest portletRequest) {
-		if ((!Objects.equals(
-				getClassName(portletRequest), DLFolder.class.getName()) &&
-			 !Objects.equals(
-				 getClassName(portletRequest), DLConstants.RESOURCE_NAME)) ||
-			_isBulkSelection(portletRequest)) {
-
-			return false;
-		}
+	protected boolean getDefaultPermissionPropagation(
+		long groupId, long classPK) {
 
-		return true;
-	}
+		if ((groupId == classPK) ||
```

Comment:
I think swap position of this to (classPK == groupId) is more readable with the second condition.

Reply:
Updated code. Thanks.